packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -360,9 +360,13 @@ export class Outbox {
 		}
 
 		const rawBatch = batchManager.popBatch(resubmittingBatchId);
-		const shouldGroup =
-			!disableGroupedBatching && this.params.groupingManager.shouldGroup(rawBatch);
-		if (batchManager.options.canRebase && rawBatch.hasReentrantOps === true && shouldGroup) {
+		const groupingEnabled =
+			!disableGroupedBatching && this.params.groupingManager.groupedBatchingEnabled();
+		if (
+			batchManager.options.canRebase &&
+			rawBatch.hasReentrantOps === true &&
+			groupingEnabled
+		) {
```

Comment:
nit: We will now rebase single message batches now even though it's not necessary, but I don't think it's a big deal

Reply:
Why isn't it necessary?

Reply:
Rebasing is only needed when we group a batch

Reply:
Turns out it breaks that random Migration jest test, so I'll restore it for now.

Reply:
I thought reentrant ops should always get rebased since the state of the world when they happened the first time could be incoherent in some since since it's in the middle of processing an incoming batch.  I think you're saying it's just about making sure an outgoing batch has a single referenceSequenceNumber.  So we should get to the bottom of it whether that first statement is a factor or not.

Reply:
In theory yes, but I don't think anything breaks if you don't rebase a reentrant ungrouped batch. But, you NEED to rebase a reentrant grouped batch.
This is because you can't use an op within a grouped batch as your reference point. You can only use the end of the grouped batch.

Reply:
I think you're saying two different things... 

> you can't use an op within a grouped batch as your reference point

This is a statement about whether the _inbound_ ops are grouped.  Which we don't have visibility into here (unless we make the "reentrancy" check smarter, which I am not really in favor of).  So I would say, this is a bug we should fix.

Reply:
Tony had a repro [here](https://github.com/microsoft/FluidFramework/compare/main...anthony-murphy:mt-rententrancy), I think it actually yielded a wrong result in Shared String, but I can't remember for sure.  Not going to dig in on this anymore for now though, the bug is tracking it.  The test failure was very difficult to investigate so punted on it once I traced it to this change.

Reply:
I think my understanding of this was incorrect, and I believe you're right that we need to always rebase. This feels like an eventual consistency bug